### PR TITLE
Add Editor page and Dashboard page feature into Researchify website

### PR DESCRIPTION
**Pull request by rpra0006**
# Description 
1. Refactored layout routes for sidebar and header. 
2. Adding a Editor.js page accessible through Dashboard to modify client webpage (functional for client home page only).
3. Dashboard page now has option to "build a website" and prompt GitHub pages token. Users are able to select desired website template when initially "building a website".
4. Fixed bug where users are able to scroll horizontally in Researchify dashboard.

# Type of change 
- [X] Bug fix
- [X] New feature
- [X] Refactoring
- [ ] Documentation 

# Problem
1. App.js has too many imports for sidebar and header layouts. Need to keep a standard layout for future implementation of new header and footer.
2. Editor page for user to customize their website pages.
3. Dashboard page for user to build an initial website.
4. See description above.

# Solution description
1. Changed the Routes in App.js into nested routes for better visibility and modification. 
See: [https://github.com/Researchify/Researchify/pull/102]
2.  Build Editor component and creating a simple Home page editor
3.  Create a simple prompt for user to enter GitHub token and selecting a template for their website
4. Added new lines in Layout.css to modify html, and body tags.

# Tests 
All unit tests pass? Yes

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

# Relevant document/ link (if any) 
Any related documents could help reviewers to review this PR 

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
High risk, please pull this branch into your branch first to see if code breaks

### CODE REVIEW

**Comment by cyr17** [comment]
Frame1 Icons should be resized as it looks stretched, and have the card container placed at the center of the screen as well
Frame 3 icons need to be replaced with words to follow consistency and they are not very informative.

So far so good, Although maybe you might have to move the personal access token to when the user registers as Akash mentioned during the last lab.

**Comment by jfan0010** [comment]
> Frame1 Icons should be resized as it looks stretched, and have the card container placed at the center of the screen as well
> Frame 3 icons need to be replaced with words to follow consistency and they are not very informative.
> 
> So far so good, Although maybe you might have to move the personal access token to when the user registers as Akash mentioned during the last lab.

@cyr17 Right the functionality of passing personal access token is not being implemented in this PR, will do another PR for it/patching this PR if manage to finish earlier. Not quite sure what the frame1/frame3 icons are referring to, would be great if you can do in-code comment :)

File: client/src/components/layouts/layoutComponents/Header.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * This function provides header for Layout.js
+ * @returns Header component to be rendered in Layout.js
+ */
+const Header = (props) => {
+    // const userName = useSelector(state => state.user?.givenName + " " + state.user?.familyName);
+    // const userName = "Albert Einstein"; // hard coded name for demo
+
+    //TODO: Remove hard-coded team id and publications id from the links
+    console.log(props)
```
Comment (cyr17) [inline client/src/components/layouts/layoutComponents/Header.js:24]: Console logs to be removed as mentioned earlier

**Review by cyr17 — APPROVED** [review]
Looks good and great work btw.

I just realized I commented on the console log which I was supposed to remove in the previous PR lol XP

File: client/src/components/dashboard/Dashboard.js [hunk 2]
```diff
@@ -1,15 +1,179 @@
...
+                            <Card.Body>
+                                <BsServer className="researchify-dashboard-card-icons" />
+                            </Card.Body>
+                            <p>API Acess Manager</p>
+                        </Link>
+                    </Card>
+                    <Card>
+                        <Link className="researchify-dashboard-card-link">
+                            <Card.Body>
+                                <BsDisplayFill className="researchify-dashboard-card-icons" />
```
Comment (cyr17) [inline client/src/components/dashboard/Dashboard.js:75]: Maybe try for a different icon for the bs server, and make these icon sizes the same, looks stretched

File: client/src/components/dashboard/Dashboard.js [hunk 3]
```diff
@@ -1,15 +1,179 @@
...
-        <main>                
-            <Container fluid>
-                You are currently viewing dashboard page. 
-            </Container>
-        </main>
+
+        <Container fluid className="researchify-dashboard-container">
+            <Card className="text-center researchify-dashboard-card">
+
+                <Card.Body>
```
Comment (cyr17) [inline client/src/components/dashboard/Dashboard.js:44]: I believe this is the body which contains the card in the darshboard, if so change the width to be adaptive and aligned to the center

File: client/src/components/dashboard/Dashboard.js [hunk 4]
```diff
@@ -1,15 +1,179 @@
...
+                    </p>
+                </Modal.Body>
+                
+                <Modal.Footer>
+                    <Button variant="secondary"
+                        onClick={() => {
+                        handleShowTwo();
+                        handleCloseThree()}}
+                    >
+                        <BsChevronLeft />
```
Comment (cyr17) [inline client/src/components/dashboard/Dashboard.js:167]: These arrow icons arent consistent and informative better to have words like usual.

**Review by sanathmahesh — APPROVED** [review]

**Review by melvinalex7 — APPROVED** [review]
Looks good! Just make sure you align the client website preview box in the dashboard, more towards the middle of the main section.

File: client/src/components/dashboard/Dashboard.js (hunk 2 above)
Comment (jfan0010) [reply client/src/components/dashboard/Dashboard.js:75]: Noted. yea i think might need to discuss later (after the demo?) on the choice of icon cause the `react-icon/bs` doesn't have much icons in it

**Pull request merged**